pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,212 @@
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 300:
+            raise ValueError("Requested voltage too large, |V| must be 300 V or less")
+         self.write("SOUR:VOLT:RAMP %g %g" % (value, time))
+
+    def ramp_to_current(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 3.5:
+            raise ValueError("Requested current too large, |I| must be 3.5 A or less")
+         self.write("SOUR:CURR:RAMP %g %g" % (value, time))
+
+    def abort_ramp(self):
+        """ Aborts all in-progress ramps
+        """
+        self.write("SOUR:CURR:RAMP:ABOR")
+        self.write("SOUR:VOLT:RAMP:ABOR")
+
+    @property
+    def source_enabled(self):
+        return self.query("OUTPut:ISOLation?")
+
+    @source_enabled.setter
+    def source_enabled(self, state):
+        if state:
+            if not self.source_enabled:
+                self.write("OUTPut:ISOLation 1")
+        elif not state:
+            if self.source_enabled:
+                if self.current_setpoint > 0:
+                    self.ramp_to_current(0,int(self.current_setpoint)*20)
+                    while self.current>0.01:
+                        time.sleep(.1)
+                    self.write("OUTPut:ISOLation 0")
+
+    def shutdown(self):
+        """ Disables the  output after ramping down at 1 V/s if voltage is non-zero."""
+        voltage = self.voltage
+        current = self.current
+        if voltage >0.1 or current >0.008:
+            self.ramp_to_voltage(0,voltage)
+            i = 0
+        while self.voltage >0.1:
+            time.sleep(1)
+            i = i+1
+            if i > 3*voltage:
+                raise ValueError("Voltage has not reached zero in reasonable time, giving up, OUTPUT IS STILL LIVE")
+        self.source_enable = False
+
+    
+
+    @property
+    def error(self):
+        """ Returns a tuple of an error code and message from a
+        single error. """
+        err = self.values(":system:error?")
+        if len(err) < 2:
+            err = self.read()  # Try reading again
+        code = err[0]
+        message = err[1].replace('"', '')
+        return (code, message)
+
+    def check_errors(self):
+        """ Logs any system errors reported by the instrument.
+        """
+        code, message = self.error
+        while code != 0:
+            t = time.time()
+            log.info("Sorensen 30035E reported error: %d, %s" % (code, message))
+            code, message = self.error
+            if (time.time() - t) > 10:
+                log.warning("Timed out for Sorensen 30035E error retrieval.")
+
```

Comment:
Could `check_errors` and `error` be removed and reuse `check_errors` from `Instrument` class ?

Reply:
yes and done